packages/langium/src/parser/langium-parser.ts

```diff
@@ -63,7 +71,7 @@ export interface BaseParser {
     /**
      * Adds a new parser rule to the parser
      */
-    rule(rule: ParserRule, impl: RuleImpl): RuleResult;
+    rule(rule: ParserRule | InfixRule, impl: RuleImpl): RuleResult;
```

Comment:
ParserRule | InfixRule occur together quite often. Maybe add a common type and use that one instead.

Reply:
That would also help for `isParserRule(rule) || isInfixRule(rule)`, which I've seen at least twice.